src/D2L.CodeStyle.Analyzers/Immutability/UnsafeStaticsAnalyzer.cs

```diff
@@ -216,50 +216,17 @@ string fieldOrPropertyName
 					)
 				);
 
-				// this bail-out isn't important because !hasDiagnostics
+				// this bail-out isn't important because !hasDiagnostic
 				return;
 			}
 
-			if ( hasAnnotations ) {
+			if ( hasAnnotation ) {
 				// the annotations supress remaining diagnostics
 				return;
 			}
 
-			while ( hasDiagnostics ) {
-				context.ReportDiagnostic( diagnostics.Current );
-				hasDiagnostics = diagnostics.MoveNext();
-			}
-		}
-
-		/// <summary>
-		/// All logic relating to either emitting or not emitting diagnostics other
-		/// than the ones about unnecessary annotations belong in this function.
-		/// This allows InspectMember to implement the logic around the unnecessary
-		/// annotations diagnostic. Any time we bail early in AnalyzeField or
-		/// AnalyzeProperty we risk not emitting unnecessary annotation
-		/// diagnostics.
-		/// </summary>
-		private IEnumerable<Diagnostic> GatherDiagnostics(

```

Comment:
Nice, yeah I was wondering about this portion when I first looked at it.

Reply:
It made sense at one point I think.